src/status_im/accounts/create/core.cljs

```diff
@@ -145,6 +111,167 @@
                               (dissoc :password :password-confirm :name :error)))}
             (navigation/navigate-to-cofx :create-account nil)))
 
+(fx/defn intro-wizard
+  {:events [:accounts.create.ui/intro-wizard]}
+  [{:keys [db] :as cofx}]
+  (fx/merge {:db (assoc db :intro-wizard {:step 1 :encrypt-with-password? true})}
+            (navigation/navigate-to-cofx :intro-wizard nil)))
+
+(fx/defn intro-step-back
+  {:events [:intro-wizard/step-back-pressed]}
+  [{:keys [db] :as cofx}]
+  (let  [step (get-in db [:intro-wizard :step])]
+    (if (< 1 step)
+      (fx/merge {:db (cond-> (assoc-in db [:intro-wizard :step] (dec step))
+                       (#{4 5} step)
+                       (assoc-in [:intro-wizard :key-code] nil)
+                       (= step 4)
+                       (assoc-in [:intro-wizard :encrypt-with-password?] false)
+                       (= step 5)
+                       (assoc-in [:intro-wizard :confirm-failure?] false))}
+                (navigation/navigate-to-cofx :intro-wizard nil))
+
+      (fx/merge {:db (dissoc db :intro-wizard)}
+                (navigation/navigate-to-clean :intro nil)))))
+
+(fx/defn intro-step-forward
+  {:events [:intro-wizard/step-forward-pressed]}
+  [{:keys [db] :as cofx} {:keys [skip?] :as opts}]
+  (let  [step (get-in db [:intro-wizard :step])
+         _ (log/info "#step-forward" step (get-in db [:intro-wizard :encrypt-with-password?])
+                     (get-in db [:intro-wizard :stored-key-code])
+                     (get-in db [:intro-wizard :key-code]))]
+
+    (cond (= step 7)
+          (fx/merge {:db (dissoc db :intro-wizard)}
+                    (when (not skip?))
+                    (navigation/navigate-to-cofx :home nil))
+          (= step 1)
+          {:db (assoc-in db [:intro-wizard :generating-keys?] true)
+           :intro-wizard/new-onboarding {:n 5 :mnemonic-length 12}}
+
+          (and (= step 5)
+               (not (:accounts/login db))
+               (get-in db [:intro-wizard :encrypt-with-password?])
+               (not= (get-in db [:intro-wizard :stored-key-code]) (get-in db [:intro-wizard :key-code])))
+          (do
+            (log/info "#step-fw confirm-failure")
+            (utils/vibrate)
+            {:db (assoc-in db [:intro-wizard :confirm-failure?] true)})
+
+          (= step 4)
+          {:db (-> db
+                   (assoc-in [:intro-wizard :stored-key-code] (get-in db [:intro-wizard :key-code]))
```

Comment:
better to use update :intro-wizard and assoc

Reply:
Thanks, yes looks better, fixed.